server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -181,22 +177,14 @@ public InternalEngine(EngineConfig engineConfig) {
                 assert translog.getGeneration() != null;
                 this.translog = translog;
                 final IndexCommit startingCommit = getStartingCommitPoint();
-                assert openMode != EngineConfig.OpenMode.OPEN_INDEX_AND_TRANSLOG || startingCommit != null :
-                    "Starting commit should be non-null; mode [" + openMode + "]; startingCommit [" + startingCommit + "]";
+                assert startingCommit != null : "Starting commit should be non-null";
```

Comment:
Just to remind you that we discussed whether to move startingCommitPoint to EngineDiskUtils.

Reply:
I decided to do this as a follow up.